accounting/src/delta/delta_data_collection/undo.rs

```diff
@@ -15,30 +15,29 @@
 
 use std::collections::BTreeMap;
 
-use super::DataDelta;
+use serialization::{Decode, Encode};
 
-/// The operations we have to do in order to undo a delta
-pub(super) enum DataDeltaUndoOpInternal<T> {
-    Write(DataDelta<T>),
-    Erase,
-}
+use super::DataDelta;
 
-pub struct DataDeltaUndoOp<T>(pub(super) DataDeltaUndoOpInternal<T>);
+#[derive(PartialEq, Eq, Clone, Encode, Decode, Debug)]
+pub struct DataDeltaUndo<T>(pub DataDelta<T>);
```

Comment:
Instead of having `pub` inside, create a function `as_delta` (which takes a reference), and also implement `From<DataDeltaUndo> for DataDelta`. I hate having to write `.0` or `.1`... nothing could look uglier and more confusing.